src/sif/autocost.cc

```diff
@@ -257,7 +257,7 @@ class AutoCost : public DynamicCost {
    * estimate is less than the least possible time along roads.
    */
   virtual float AStarCostFactor() const override {
-    return speedfactor_[kMaxAssumedSpeed];
+    return speedfactor_[top_speed_];
```

Comment:
`top_speed_` always defaults to `kMaxSpeedKph` (if not explicitly set) in all (but scooter) motorized vehicles. before some cost models used `kMaxAssumedSpeed`, others `kMaxSpeedKph`. not sure if this is a good change..

Reply:
its not a good change as it will impact the astar heuristic in such a way as to assume a given node is closer to the goal (because you can travel faster). this will slow down the algorithm for sure. and...

just tested it, its about 17% slower which is defintiely not acceptable. we'll want to revert that change for sure

Reply:
my bad.. I might've only "seen" the previous truck implementation, which assumed `kMaxSpeedKph` here. My goal was actually the other way around, to make astar go faster (when a top_speed is actually set) :sweat_smile: Clamping the `top_speed` to `kMaxAssumedSpeed` will probably not be desirable (I guess since there's the `kMaxTrafficSpeed` of 252)?!

Reply:
this is what I should've done... what'ya think?
```
  virtual float AStarCostFactor() const override {
    uint32_t s = top_speed_ < kMaxAssumedSpeed ? top_speed_ : kMaxAssumedSpeed;
    return speedfactor_[s];
  }
```